village/src/components/SmurfForm/SmurfForm.js

```diff
@@ -1,54 +1,63 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+
+import "./SmurfForm.css";
 
 class SmurfForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      name: '',
-      age: '',
-      height: ''
+      name: "",
+      age: "",
+      height: ""
     };
   }
 
   addSmurf = event => {
     event.preventDefault();
     // add code to create the smurf using the api
-
+    this.props.addSmurf(this.state);
```

Comment:
Good use of passing props and keeping your components clean. This is a good function for adding data.